src/Mvc/src/Microsoft.AspNetCore.Mvc.Razor/Compilation/CompiledViewDescriptor.cs

```diff
@@ -80,5 +82,23 @@ public CompiledViewDescriptor(RazorCompiledItem item, RazorViewAttribute attribu
         /// Gets the type of the compiled item.
         /// </summary>
         public Type Type => Item?.Type ?? ViewAttribute?.ViewType;
+
+        internal Func<IRazorPage> CreatePageFactory()
+        {
+            Debug.Assert(typeof(IRazorPage).IsAssignableFrom(Type));
+
+            var newExpression = Expression.New(Type);
```

Comment:
Worth validating here that the razor page type has a default constructor?

Reply:
Doesn't compile do that?

Reply:
That would make sense and it does. Never mind.